version1/04 - Web services.md

```diff
@@ -468,6 +468,128 @@ Content-Type: */*
+| Korrespondansepart | Korrespondansepart                                                                                                                                                                                                                                                                                                     |
+| Registrering       | AbstraktRegistrering (all types), Noekkelord, Korrespondansepart, virksomhetsspesifikkeMetadata, Klasse (secondary), Klassifikasjonssystem (secondary), AbstraktMappe, Arkivdel, Arkiv                                                                                                                                 |
+| Mappe              | AbstraktMappe (all types), Noekkelord, Sakspart, virksomhetsspesifikkeMetadata                                                                                                                                                                                                                                         |
+| Arkivdel           | Arkivdel                                                                                                                                                                                                                                                                                                               |
+
+## **search**
+
+Performs free text search in index documents.
+
+**Request**
+
+``` text
+POST /rms/api/public/noark5/v1/full-text/search HTTP/1.1
+Authorization: Bearer ACCESS_TOKEN
+Content-Type: application/json
+
+{
+  "doctype": string,
+  "query": string,
+  "offset: number,
+  "limit": number
+}
+```
+
+###### Details
+
+- **doctype**
+  - search for index documents with the specified doctype
+- **query**
+  - free text search string
+  - the search is performed across multiple fields
+  - the default search fields in the corresponding object types are:
+    - Klasse: klasseIdent, tittel
+    - AbstraktMappe: mappeIdent, tittel, offentligTittel, saksansvarlig
+    - Sakspart: sakspartNavn, kontaktperson
+    - AbstraktRegistrering: registreringsIdent, tittel, offentligTittel
+    - Korrespondansepart: korrespondansepartNavn, epostadresse, kontaktperson, foedselsnummer
+    - Dokument: tittel, beskrivelse
+    - Tekst (of the last Dokumentversjon)
+    - Noekkelord: verdi
+    - Arkivdel: tittel
+- **offset** (optional)
+  - offset of the first result to retrieve
+  - defaults to *0*
+- **limit** (optional)
+  - maximum number of results to retrieve
+  - defaults to *10*
+
+**Response**
+
+``` text
+Content-Type: application/json
+
+{
+  "total": number,
+  "results": [
+    {
+      "ids": {
+        string: [number],
+        ...
+      },
+      "highlights": {
+        string: [string],
+        ...
+      }
+    },
+    ...
+  ],
+  "facets": [
+    {
+      "field": string,
+      "values": {
+        string:number,
+        ...
+      }
+    },
+    ...
+  ]
+}
+```
+
+**Details**
```

Comment:
I would also add a comment to `facets` that other facet fields may be returned too and clients must not make an assumption about the exact set of facets returned by this endpoint.

Reply:
Added.